File: src/org/usfirst/frc5293/prefs/util/BooleanPref.java
package org.usfirst.frc5293.prefs.util;

import edu.wpi.first.wpilibj.Preferences;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class BooleanPref extends Pref<Boolean> {
    public BooleanPref(String key, Boolean defaultValue) {
        super(key, defaultValue);
    }

    @Override
    public Boolean get() {
        return Preferences.getInstance().getBoolean(key, defaultValue);
    }

    @Override
    public void set(Boolean value) {
        Preferences.getInstance().putBoolean(key, value);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/prefs/util/BooleanPref.kt
package org.usfirst.frc5293.prefs.util

import edu.wpi.first.wpilibj.Preferences

class BooleanPref(key: String, default: Boolean) : Pref<Boolean>(key, default) {

    override fun get(): Boolean {
        return Preferences.getInstance().getBoolean(key, default)
    }

    override fun set(value: Boolean) {
        Preferences.getInstance().putBoolean(key, value)
    }
}
